apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/SystemInfoController.java

```diff
@@ -84,8 +68,52 @@ public SystemInfo getSystemInfo() {
 
   @PreAuthorize(value = "@permissionValidator.isSuperAdmin()")
   @GetMapping(value = "/health")
-  public Health checkHealth(@RequestParam String host) {
-    return restTemplate.getForObject(host + "/health", Health.class);
+  public Health checkHealth(@RequestParam String instanceId) {
+    List<Env> allEnvs = portalSettings.getAllEnvs();
+
+    ServiceDTO service = null;
+    for (final Env env : allEnvs) {
+      EnvironmentInfo envInfo = adaptEnv2EnvironmentInfo(env);
+      for (final ServiceDTO s : envInfo.getAdminServices()) {
+        if (instanceId.equals(s.getInstanceId())) {
+          service = s;
+          break;
+        }
+      }
+      for (final ServiceDTO s : envInfo.getConfigServices()) {
+        if (instanceId.equals(s.getInstanceId())) {
+          service = s;
+          break;
+        }
+      }
+    }
+
+    if (service == null) {
+      throw new IllegalArgumentException("No such instance of instanceId: " + instanceId);
+    }
+
+    return restTemplate.getForObject(service.getHomepageUrl() + "/health", Health.class);
+  }
+
+  private EnvironmentInfo adaptEnv2EnvironmentInfo(final Env env) {
+    EnvironmentInfo environmentInfo = new EnvironmentInfo();
+    String metaServerAddresses = MetaDomainConsts.getMetaServerAddress(env);
+
+    environmentInfo.setEnv(env);
+    environmentInfo.setActive(portalSettings.isEnvActive(env));
+    environmentInfo.setMetaServerAddress(metaServerAddresses);
+
+    String selectedMetaServerAddress = MetaDomainConsts.getDomain(env);
+    try {
+      environmentInfo.setConfigServices(getServerAddress(selectedMetaServerAddress, CONFIG_SERVICE_URL_PATH));
+
+      environmentInfo.setAdminServices(getServerAddress(selectedMetaServerAddress, ADMIN_SERVICE_URL_PATH));
+    } catch (Throwable ex) {
+      String errorMessage = "Loading config/admin services from meta server: " + selectedMetaServerAddress + " failed!";
+      logger.error(errorMessage, ex);
+      environmentInfo.setErrorMessage(errorMessage + " Exception: " + ex.getMessage());
```

Comment:
In this case, environmentInfo.getAdminServices and environmentInfo.getConfigServices will return null and there will be null pointer exception in line 77 and line 83